doc/Operation.md

```diff
@@ -0,0 +1,328 @@
+# Operation\<T>
+
+The Azure SDK for C++ defines an Operation as an abstract class to support a long running operation ([LRO](http://restalk-patterns.org/long-running-operation-polling.html)).
+
+The SDK clients would define a concrete class from the `Operation<T>` which an end users can consume for getting the status of the running operation.
+
+## Operation Status
+
+The status of the operation defines the stages which the operation can be set from its creation and up to completion. The next table show the supported status of an `Operation<T>`.
+
+| Status     | Description                                                                    |
+| ---------- | ------------------------------------------------------------------------------ |
+| NotStarted | The operation is created but the server has not yet started it.                |
+| Running    | The operation is running on the server side and has not yet be completed.      |
+| Succeeded  | The server has successfully finished running the LRO. This is a `final stage`. |
+| Cancelled  | The operation was interrupted before completion. This is a `final stage`.      |
+| Failed     | The server failed to run the LRO. This is a `final stage`.                     |
+
+> Note: A final stage means that the status won't change anymore.
+
+## Consuming an Operation\<T>
+
+Any concrete class of `Operation<T>` will expose a `Poll()` and `PollUntilDone()` APIs. These methods will update the `operation status`. The status of an operation won't be automatically updated without calling `Poll()`.
+
+### Poll
+
+The `Poll()` method will send a request to the Server to get the current status of the `LRO`. The Server will then return an Http raw response with information about the completion of the LRO. The Operation parses the response and updates its status. Finally, a read only reference to the http raw response (the one used to update the status) is returned by `Poll()`. We can use this raw response to parse specific information contained in the response such a progress, time, percentage or other data. The operation will only check for changes in the `operation status`, but the operation-consumer will know better about all the data coming back from the server in the response.
+
+The next code demonstrates the use of `Poll()`.
+
+```cpp
+/**
+ * Consuming Operation<T> interface to Poll for LRO updates.
+ *
+ */
+auto operationResult = client.ApiReturnOperation(...);
+
+while (true)
```

Comment:
using `while(operation.IsDone())` is not convenience because calling `Poll` updates the m_status, se we need to check `IsDone()` after calling `Poll` which would duplicate the condition from the while and inside the while